addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/environments-sc/parts/ScEnvironmentSsmConnections.js

```diff
@@ -0,0 +1,158 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+import _ from 'lodash';
+import React from 'react';
+import { decorate, computed, action, runInAction, observable } from 'mobx';
+import { observer, inject } from 'mobx-react';
+import { withRouter } from 'react-router-dom';
+import { Segment, Icon, Button, Header, Table } from 'semantic-ui-react';
+
+import { displayError } from '@amzn/base-ui/dist/helpers/notification';
+
+const openWindow = (url, windowFeatures) => {
+  return window.open(url, '_blank', windowFeatures);
+};
+
+class ScEnvironmentSsmConnections extends React.Component {
+  constructor(props) {
+    super(props);
+    runInAction(() => {
+      this.processingId = '';
+      this.timeout = 10;
+    });
+  }
+
+  get environment() {
+    return this.props.scEnvironment;
+  }
+
+  get envsStore() {
+    return this.props.scEnvironmentsStore;
+  }
+
+  getConnectionStore() {
+    return this.envsStore.getScEnvConnectionStore(this.environment.id);
+  }
+
+  get connections() {
+    const isSsm = scheme => scheme === 'ssm';
+    const connections = this.environment.getConnections(item => isSsm(item.scheme));
+    return connections;
+  }
+
+  handleConnect = id =>
+    action(async () => {
+      const store = this.getConnectionStore();
+      const connections = this.environment.connections;
+      const connectInfo = _.find(connections, ['id', id]) || {};
```

Comment:
prevent opening more than one SSM connect window? does it work?

Reply:
Currently, we cannot prevent opening more than one window,  implementation is the same with Sagemaker.
Allow user to open multi windows